### PR TITLE
Fix TextBox NullReferenceException on unsupported characters

### LINKED ISSUE #1740: TextBox throws on newline character
Specifically the _CR and LF_ characters.

Steps to reproduce:

1. Open notepad and type stuff that has multiple lines
2. Copy pasta into TextBox
3. Exception!

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/61433714-textbox-throws-on-newline-character?utm_campaign=plugin&utm_content=tracker%2F44024461&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F44024461&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by phosphene47**
- Resolves #1740 

---

### CODE REVIEW

**Comment by peppy** [comment]
Was there a reason for not just adding a null check, rather than using an `out` parameter?

File: osu.Framework/Graphics/UserInterface/TextBox.cs [hunk 1]
```diff
@@ -503,17 +503,20 @@ private void insertString(string addText)
...
+                if (!addCharacter(c, out var ch))
+                    continue;
 
                 var col = (Color4)ch.Colour;
                 ch.FadeColour(col.Opacity(0)).FadeColour(col, caret_move_time * 2, Easing.Out);
             }
         }
 
-        private Drawable addCharacter(char c)
+        private bool addCharacter(char c, out Drawable drawableChar)
```
Comment (UselessToucan) [inline osu.Framework/Graphics/UserInterface/TextBox.cs]: Should be named `tryAddCharacter`.

File: osu.Framework/Graphics/UserInterface/TextBox.cs [hunk 2]
```diff
@@ -503,17 +503,20 @@ private void insertString(string addText)
 
             foreach (char c in addText)
             {
-                var ch = addCharacter(c);
+                if (!addCharacter(c, out var ch))
+                    continue;
```
Comment (UselessToucan) [inline osu.Framework/Graphics/UserInterface/TextBox.cs]: Is it ok to ignore the fact that we did not perform the insertion?

**Comment by smoogipoo** [comment]
Superseded by https://github.com/ppy/osu-framework/pull/1778

**Pull request closed without merging**